test/hooks/test-trace-koa.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var common = require('./common.js');
+var koa = require('./fixtures/koa1');
+
+var server;
+
+describe('test-trace-koa', function() {
+  afterEach(function() {
+    common.cleanTraces();
+    server.close();
+  });
+
+  it('should accurately measure get time, get', function(done) {
+    var app = koa();
+    app.use(function *() {
+      this.body = yield function(cb) {
+        setTimeout(function() {
+          cb(null, common.serverRes);
+        }, common.serverWait);
+      };
+    });
+    server = app.listen(common.serverPort, function() {
+      common.doRequest('GET', done, koaPredicate);
+    });
+  });
+});
+
```

Comment:
We currently do not have this type of testing for any of our frameworks and discussed incorporating it into the e2e tests. The method, url, ip, status_code, agent, hostname, and app/module labels have been verified manually. I can open a separate bug for this across all the frameworks if you'd like.